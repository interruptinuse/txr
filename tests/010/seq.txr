@(do 
   (defvar *h* (hash :equal-based))
   (defvar *s* "abcd")
   (defvar *v* #(1 2 3 4 5 6 7 8))
   (defvar *v2* #((1 . #\a) (2 . #\b) (3 . #\c) (4 . #\d)
                  (5 . #\e) (6 . #\f) (7 . #\h) (8 . #\g)))
   (defun caught (x) (format t "exception!\n"))
   (defun pr (x) (format t "~s\n" x))
   (defun pr2 (x y) (format t "~s ~s\n" x y))
   (pr (set [*h* "a"] 42))
   (pr [*h* "a"])
   (pr (del [*h* "a"]))
   (pr (del [*h* "a"]))
   (format t "~s ~s\n" (del [*s* 1]) *s*)
   (format t "~s ~s\n" (del [*s* -1]) *s*)
   (catch (pr (del [*s* 3]) *s*) (t (x) (caught x)))
   (pr [nsort *v* >])
   (pr [nsort *v2* > cdr])
   (pr [nsort (range 1 100) >])
   (pr2 (del [*v2* 1..3]) *v2*)
   (let ((s (copy "abcdefg")))
     (pr (del [s '(1 3 5)]))
     (pr s))
   (let ((s (copy "abcdefg")))
     (pr (del [s '(6)]))
     (pr s))
   (let ((s (copy "abcdefg")))
     (pr (del [s '(0 1 2 3 4 5 6)]))
     (pr s))
   (let ((s (copy "abcdefg")))
     (pr (del [s '(-7 -6 -5 -4 -3 -2 -1)]))
     (pr s))
   (let ((s (copy "abcdefg")))
     (pr (del [s '(-7 -5 -3 -1)]))
     (pr s))
   (let ((s (list 0 1 2 3 4 5 6)))
     (pr (del [s '(1 3 5)]))
     (pr s))
   (let ((s (list 0 1 2 3 4 5 6)))
     (pr (del [s '(0 1 2 3 4 5 6)]))
     (pr s))
   (let ((s (list 0 1 2 3 4 5 6)))
     (pr (del [s '(-7 -6 -5 -4 -3 -2 -1)]))
     (pr s))
   (let ((s (list 0 1 2 3 4 5 6)))
     (pr (del [s '(-7 -5 -3 -1)]))
     (pr s))
   (let ((s (vec 0 1 2 3 4 5 6)))
     (pr (del [s '(1 3 5)]))
     (pr s))
   (let ((s (vec 0 1 2 3 4 5 6)))
     (pr (del [s '(0 1 2 3 4 5 6)]))
     (pr s))
   (let ((s (vec 0 1 2 3 4 5 6)))
     (pr (del [s '(-7 -6 -5 -4 -3 -2 -1)]))
     (pr s))
   (let ((s (vec 0 1 2 3 4 5 6)))
     (pr (del [s '(-7 -5 -3 -1)]))
     (pr s))
   (let ((s (copy #b'aabbccddeeff99')))
     (pr (del [s '(1 3 5)]))
     (pr s))
   (let ((s (copy #b'aabbccddeeff99')))
     (pr (del [s '(6)]))
     (pr s))
   (let ((s (copy #b'aabbccddeeff99')))
     (pr (del [s '(0 1 2 3 4 5 6)]))
     (pr s))
   (let ((s (copy #b'aabbccddeeff99')))
     (pr (del [s '(-7 -6 -5 -4 -3 -2 -1)]))
     (pr s))
   (let ((s (copy #b'aabbccddeeff99')))
     (pr (del [s '(-7 -5 -3 -1)]))
     (pr s))
   )
