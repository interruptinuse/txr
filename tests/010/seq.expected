42
42
42
nil
#\b "acd"
#\d "ac"
exception!
#(8 7 6 5 4 3 2 1)
#((7 . #\h) (8 . #\g) (6 . #\f) (5 . #\e) (4 . #\d) (3 . #\c) (2 . #\b)
  (1 . #\a))
(100 99 98 97 96 95 94 93 92 91 90 89 88 87 86 85 84 83 82 81
 80 79 78 77 76 75 74 73 72 71 70 69 68 67 66 65 64 63 62 61 60
 59 58 57 56 55 54 53 52 51 50 49 48 47 46 45 44 43 42 41 40 39
 38 37 36 35 34 33 32 31 30 29 28 27 26 25 24 23 22 21 20 19 18
 17 16 15 14 13 12 11 10 9 8 7 6 5 4 3 2 1)
#((8 . #\g) (6 . #\f)) #((7 . #\h) (5 . #\e) (4 . #\d) (3 . #\c) (2 . #\b) (1 . #\a))
"bdf"
"aceg"
"g"
"abcdef"
"abcdefg"
""
"abcdefg"
""
"aceg"
"bdf"
(1 3 5)
(0 2 4 6)
(0 1 2 3 4 5 6)
nil
(0 1 2 3 4 5 6)
nil
(0 2 4 6)
(1 3 5)
#(1 3 5)
#(0 2 4 6)
#(0 1 2 3 4 5 6)
#()
#(0 1 2 3 4 5 6)
#()
#(0 2 4 6)
#(1 3 5)
#b'bbddff'
#b'aaccee99'
#b'99'
#b'aabbccddeeff'
#b'aabbccddeeff99'
#b''
#b'aabbccddeeff99'
#b''
#b'aaccee99'
#b'bbddff'
